native/cocos/physics/physx/joints/PhysXJoint.cpp

```diff
@@ -69,16 +69,32 @@ void PhysXJoint::setConnectedBody(uint32_t rigidBodyID) {
     if (pxRigidBody == nullptr)
         return;
 
+
+    auto *oldConnectedBody = _mConnectedBody;
+    if (oldConnectedBody) {
+        oldConnectedBody->removeJoint(*this, physx::PxJointActorIndex::eACTOR1);
+    }
+
     uintptr_t nodePtr = reinterpret_cast<uintptr_t>(pxRigidBody->getSharedBody().getNode());
     if (nodePtr) {
         auto &ins = PhysXWorld::getInstance();
         _mConnectedBody = ins.getSharedBody(reinterpret_cast<Node *>(nodePtr));
+        _mConnectedBody->addJoint(*this, physx::PxJointActorIndex::eACTOR1);
     } else {
         _mConnectedBody = nullptr;
     }
     if (_mJoint) {
         _mJoint->setActors(_mSharedBody->getImpl().rigidActor, _mConnectedBody ? _mConnectedBody->getImpl().rigidActor : nullptr);
     }
+
+    if (oldConnectedBody) {
+        if (oldConnectedBody->isDynamic()) {
+            oldConnectedBody->getImpl().rigidDynamic->wakeUp();
+        }
+    }
+
+    updateScale0();
```

Comment:
updateScale0() and updateScale1() should have more specific name

Reply:
`updateScale0` and `updateScale1` are inherited from their parents. We could consider refactoring them in the future.